integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,259 @@
+  private final Optional<String> mFuseAuthPolicyCustomUser;
+  private final boolean mUserGroupTranslationEnabled;
+  // libfuse-related options
+  private final List<String> mLibfuseOptions;
+  // general options
+  private final boolean mIsDebug;
+
+  /**
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with only Alluxio cluster configuration.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(AlluxioConfiguration conf) {
+    Preconditions.checkNotNull(conf,
+        "AlluxioConfiguration should not be null for creating AlluxioFuseFileSystemOpts.");
+    return create(conf, AlluxioFuseCliOpts.empty());
+  }
+
+  /**
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with
+   * Alluxio cluster configuration and command line input.
+   * Command line input has higher precedence if a property is set both in config and command.
+   *
+   * @param conf Alluxio cluster configuration
+   * @param fuseCliOpts Alluxio fuse command line input
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(
+      AlluxioConfiguration conf, AlluxioFuseCliOpts fuseCliOpts) {
+    Preconditions.checkNotNull(conf,
+        "AlluxioConfiguration should not be null for creating AlluxioFuseFileSystemOpts.");
+    Preconditions.checkNotNull(fuseCliOpts,
+        "AlluxioFuseCliOpts should not be null for creating AlluxioFuseFileSystemOpts.");
+    String alluxioPath = fuseCliOpts.getMountAlluxioPath().orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH));
+    Optional<List<String>> libfuseOptionsFromCli = fuseCliOpts.getFuseOptions();
+    List<String> libfuseOptions;
+    if (libfuseOptionsFromCli.isPresent()) {
+      libfuseOptions = libfuseOptionsFromCli.get();
+    } else {
+      if (conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)) {
+        libfuseOptions = conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS);
+      } else {
+        libfuseOptions = ImmutableList.of();
+      }
+    }
+    libfuseOptions = optimizeAndFormatFuseOptions(libfuseOptions);
+    String mountPoint = fuseCliOpts.getMountPoint().orElseGet(
+        () -> conf.getString(PropertyKey.FUSE_MOUNT_POINT));
+    Optional<String> authPolicyCustomGroup = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP)
+        && !conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP).isEmpty()
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_GROUP))
+        : Optional.empty();
+    Optional<String> authPolicyCustomUser = conf.isSet(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER)
+        && !conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER).isEmpty()
+        ? Optional.of(conf.getString(PropertyKey.FUSE_AUTH_POLICY_CUSTOM_USER))
+        : Optional.empty();
+    return new AlluxioFuseFileSystemOpts(alluxioPath,
+        conf.getString(PropertyKey.FUSE_FS_NAME),
```

Comment:
are we 100% sure that those operation wont throw runtime exception, or if it throws, do we handle them well?

Reply:
You mean the `conf.getXxx`? If they can throw runtime exception, our codebase will be in hell that everywhere we get something from configuration we need to take care of exception.

Reply:
They can definitely throw RuntimeException with the assumption that if a RuntimeException is throw, the system should be shut down or the exception be thrown to end-user.

Reply:
They can definitely throw RuntimeException with the assumption that if a RuntimeException is throw, the system should be shut down or the exception be thrown to end-user.

Reply:
Currently we don't handle the runtime exceptions, so fuse won't be able to launch and the exception will be logged into `fuse.out`